doc/api.rst

```diff
@@ -0,0 +1,478 @@
+
+        :param shape: The shape of the DPMF, as its integer length.
+        :type shape: int
+        :returns: DPMF with given shape and uniform probabilities.
+        :rtype: DPMF
+
+Usage example:
+
+.. code-block:: python
+
+    # Create a custom DPMF object in the InputMaster mean InlineSection
+    input_master.dpmf["biased"] = DPMF([0.7, 0.3])
+    # Create a uniform DPMF with a specified shape
+    input_master.dpmf["uniform"] = DPMF.uniform_from_shape(3)
+
+
+.. py:class:: DiagGaussianMC
+
+    A Gaussian distribution with diagonal covariance. Currently the only 
+    concrete MC subclass which can be used as a mixture component. 
+
+    .. py:attribute:: mean
+        :type: str
+
+        Name of a :py:class:`Mean` object representing the mean of this 
+        Gaussian.
+    
+    .. py:attribute:: covar
+        :type: str
+        
+        Name of a :py:class:`Covar` object representing the covariance 
+        vector along the diagonal of the covariance matrix. 
+
+    .. py:method:: __init__(self, mean, covar)
+
+        Create a :py:class:`DiagGaussianMC` object with the specified mean 
+        and covariance.
+
+        :param mean: Name of a Mean object for the distribution mean
+        :type mean: str
+        :param covar: Name of a Covar object for the diagonal covariance vector of the distribution
+        :type covar: str
+
+Usage example:
+
+.. code-block:: python
+
+    # Create a DiagGaussian object in the InputMaster mc 
+    # InlineMCSection.
+    # Arguments are labels for Mean and Covariance objects.
+    input_master.mc["dist1"] = \
+        DiagGaussianMC(mean = "dist1", covar = "dist1")
+    input_master.mc["dist2"] = \
+        DiagGaussianMC(mean = "dist2", covar = "dist2")
+
+
+.. py:class:: MX
+
+    A Gaussian mixture model built from Gaussian mixture components.
+
+    .. py:attribute:: dpmf
+        :type: str
+
+        Name of a :py:class:`DPMF` object representing the contribution of 
+        each Gaussian mixture component to the mixture model.
+    
+    .. py:attribute:: components
+        :type: str or list[str]
+
+        Names of Gaussian components associated with the mixture model. 
+
+    .. py:method:: __init__(self, dpmf, components)
+
+        Create an :py:class:`MX` object with the mixture distribution and 
+        components.
+
+        :param dpmf: Name of a DPMF describing mixture weights.
+        :type dpmf: str
+        :param components: Name or list of names of mixture components
+        :type components: str or list[str]
+
+Usage example:
+
+.. code-block:: python
+
+    # Create a MX objects in the InputMaster mx InlineMXSection.
+    # Arguments are labels for DPMF and MX objects.
+    input_master.mx["emission1"] = MX("biased", ["dist1", "dist2"])
+    input_master.mx["emission2"] = MX("biased", ["dist1", "dist2"])
+
+
+.. py:class:: DenseCPT
+
+    A Numpy ``ndarray`` representing a dense CPT with a specialized string method 
```

Comment:
Again expand the CPT acronym at least once and anywhere else in the docs for other acronyms that show up.